Userland/Libraries/LibWeb/HTML/HTMLScriptElement.cpp

```diff
@@ -434,13 +440,20 @@ void HTMLScriptElement::prepare_script()
         }
         // -> "importmap"
         else if (m_script_type == ScriptType::ImportMap) {
-            // FIXME: 1. If el's relevant global object's import maps allowed is false, then queue an element task on the DOM manipulation task source given el to fire an event named error at el, and return.
-
-            // FIXME: 2. Set el's relevant global object's import maps allowed to false.
-
-            // FIXME: 3. Let result be the result of creating an import map parse result given source text and base URL.
-
-            // FIXME: 4. Mark as ready el given result.
+            // 1. If el's relevant global object's import maps allowed is false, then queue an element task on the DOM manipulation task source given el to fire an event named error at el, and return.
+            if (!document().window().import_maps_allowed()) {
+                queue_an_element_task(HTML::Task::Source::DOMManipulation, [this] {
+                    dispatch_event(DOM::Event::create(realm(), HTML::EventNames::error));
+                });
+                return;
+            }
+            // 2. Set el's relevant global object's import maps allowed to false.
+            document().window().set_import_maps_allowed(false);
+            // 3. Let result be the result of creating an import map parse result given source text and base URL.
+            auto result = parse_an_import_map_string()
```

Comment:
Please do pass `source_text` and `base_url` as parameters to this, and the other spec algorithms that you're adding here.